### PR TITLE
Dockerfile.headnode

### PR DESCRIPTION
**Pull request by solaris-x86**
Please download the dockerfile and build the image by "docker build -t hadoop:headnode -f Dockerfile.headnode" command.

Then you can run the container by "docker run -d --name=headnode --hostname=headnode --network=hadoopnet hadoop:headnode"

Also you can pull and run the builded image from Docker hub by "docker run -d --name=headnode --hostname=headnode --network=hadoopnet mmmain/hadoop:headnode"

@Cakemanator
@yegorgeorgiyev
@Shlepanets
@nikita-safonov

### CODE REVIEW

**Comment by solaris-x86** [comment]
Also you can open the Dockerfile for "worker" container https://raw.githubusercontent.com/MikhYur1991/devOPSp2/Docker/Dockerfile.worker

Please download this dockerfile and build the image by "docker build -t hadoop:worker -f Dockerfile.worker" command.

Then you can run the container by "docker run -d --name=worker --hostname=worker --network=hadoopnet hadoop:worker"

Also you can pull and run the builded image from Docker hub by "docker run -d --name=worker --hostname=worker --network=hadoopnet mmmain/hadoop:worker"

**Comment by solaris-x86** [comment]
Also you can download docker-compose file https://raw.githubusercontent.com/MikhYur1991/devOPSp2/Docker/compose.yml and start both containers by "docker-compose up -d" command without any parameters

File: compose.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+version: "3.9"
```
Comment (Cakemanator) [inline compose.yml:1]: What can we add to the compose file to make images build automatically?

File: compose.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+services:
+   headnode:
+      image: mmmain/hadoop:headnode
+      container_name: headnode
+      volumes:
+         - hmnt1:/opt/mount1
+         - hmnt2:/opt/mount2
+         - hlogs:/usr/local/hadoop/current/logs
+#      ports:
```
Comment (Cakemanator) [inline compose.yml:10]: Should we uncomment this in order to allow users to easily reach WebUI?

**Review by Cakemanator — COMMENTED** [review]
Approved with additional questions

**Review by Cakemanator — APPROVED** [review]

File: compose.yml (hunk 2 above)
Comment (solaris-x86) [reply compose.yml:10]: We can use a "localhost:8088" in the browser address field in this case (uncommented "ports" option)

File: compose.yml (hunk 1 above)
Comment (solaris-x86) [reply compose.yml:1]: We have to add the "build" option with its context. For example, 
version: "3.9"
services:
   headnode2:
      build:
         context: https://raw.githubusercontent.com/MikhYur1991/devOPSp2/Docker/Dockerfile.headnode
      volumes:
         - h2mnt1:/opt/mount1
         - h2mnt2:/opt/mount2
         - h2logs:/usr/local/hadoop/current/logs
      networks:
         - hadoopnet
volumes:
   h2mnt1:
      name: h2mount1
   h2mnt2:
      name: h2mount2
   h2logs:
      name: headnode2logs
networks:
   hadoopnet:
      driver: bridge
      name: hadoopnet

**Pull request merged**